root/static/scripts/edit/utility/linkPhrase.js

```diff
@@ -263,9 +261,13 @@ export function getPhraseAndExtraAttributes<T>(
     }
   }
 
-  const requiredAttributes = _getRequiredAttributes(
-    linkType,
-    attributesByRootName,
+  const requiredAttributes = Object.fromEntries(
+    Object.entries(_getRequiredAttributes(linkType)).map(
```

Comment:
Probably stupid question: what's the use of taking object.entries and then making object.fromEntries here?

Reply:
It's mapping over the keys to create a new object with different values--in this case, the passed-in attributes for a specific relationship. (There's no equivalent of `Array.prototype.map` for objects.)